src/model/test/CoilCoolingDXSingleSpeedThermalStorage_GTest.cpp

```diff
@@ -848,12 +859,103 @@ TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_HeatCoolFuelTypes) {
+  Model m;
+
+  CoilCoolingDXSingleSpeedThermalStorage dx(m);
+
+  AirLoopHVAC airLoop(m);
+
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+  EXPECT_FALSE(dx.airLoopHVAC());
+  EXPECT_EQ((unsigned)2, airLoop.supplyComponents().size());
+  EXPECT_TRUE(dx.addToNode(supplyOutletNode));
+  EXPECT_EQ((unsigned)3, airLoop.supplyComponents().size());
+  EXPECT_TRUE(dx.airLoopHVAC());
+
+  Node demandNode = airLoop.zoneSplitter().lastOutletModelObject()->cast<Node>();
+  EXPECT_FALSE(dx.addToNode(demandNode));
+  // 5u: inlet splitter node mixer outlet.
+  EXPECT_EQ((unsigned)5, airLoop.demandComponents().size());
+  EXPECT_EQ((unsigned)3, airLoop.supplyComponents().size());
+  EXPECT_TRUE(dx.airLoopHVAC());
+
+  EXPECT_TRUE(dx.removeFromLoop());
+  EXPECT_EQ((unsigned)2, airLoop.supplyComponents().size());
+  EXPECT_FALSE(dx.airLoopHVAC());
+
+  PlantLoop plantLoop(m);
+  supplyOutletNode = plantLoop.supplyOutletNode();
+  EXPECT_FALSE(dx.addToNode(supplyOutletNode));
+  EXPECT_EQ((unsigned)5, plantLoop.supplyComponents().size());
+
+  Node demandOutletNode = plantLoop.demandOutletNode();
+  EXPECT_FALSE(dx.addToNode(demandOutletNode));
+  EXPECT_EQ((unsigned)5, plantLoop.demandComponents().size());
+}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_clone) {
+  Model model;
+
+  CoilCoolingDXSingleSpeedThermalStorage dx(model);
+
+  EXPECT_EQ(1u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+
+  auto dxClone = dx.clone(model).cast<CoilCoolingDXSingleSpeedThermalStorage>();
+  EXPECT_EQ(2u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+
+  dx.remove();
+  EXPECT_EQ(1u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+
+  dxClone.remove();
+  EXPECT_EQ(0u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+}
+
+TEST_F(ModelFixture, CoilCoolingDXSingleSpeedThermalStorage_cloneParent) {
+  Model model;
+
+  CoilCoolingDXSingleSpeedThermalStorage dx(model);
+
+  AirLoopHVACUnitarySystem unitary(model);
+  EXPECT_TRUE(unitary.setCoolingCoil(dx));
+
+  ASSERT_TRUE(dx.containingHVACComponent());
+  EXPECT_EQ(unitary, dx.containingHVACComponent().get());
+
+  ASSERT_TRUE(unitary.coolingCoil());
+  EXPECT_EQ(dx, unitary.coolingCoil().get());
+
+  EXPECT_EQ(1u, model.getConcreteModelObjects<AirLoopHVACUnitarySystem>().size());
+  EXPECT_EQ(1u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+
+  // Cloning the unitary should clone the coolingCoil, but the clone of the DX should still share resources
+  auto unitaryClone = unitary.clone(model).cast<AirLoopHVACUnitarySystem>();
+  EXPECT_EQ(2u, model.getConcreteModelObjects<AirLoopHVACUnitarySystem>().size());
+  EXPECT_EQ(2u, model.getConcreteModelObjects<CoilCoolingDXSingleSpeedThermalStorage>().size());
+  EXPECT_EQ(dx, unitary.coolingCoil().get());
+  EXPECT_NE(dx, unitaryClone.coolingCoil().get());
```

Comment:
Need test to check what happens when you clone to another model. And when removing (are curves dragged / removed?)